test/modules/telemetry/telemetry.js

```diff
@@ -0,0 +1,34 @@
+import { readFile } from 'fs/promises';
+import { describe, it, before } from 'mocha';
+import { assert } from 'chai';
+import Logger from '../../../src/logger/logger.js';
+import TelemetryModuleManager from '../../../src/modules/telemetry/telemetry-module-manager.js';
+
+let logger;
+let telemetryModuleManager;
+const config = JSON.parse(await readFile('./test/modules/telemetry/config.json'));
+
+describe('Telemetry module', () => {
+    before('Initialize telemetry module manager', () => {
+        logger = new Logger('trace');
+        logger.info = () => {};
+    });
+
+    describe('Listen on events', () => {
+        it('Event emitter on method called', async () => {
+            const eventEmitterMock = {
+                on: (eventName) => {
+                    assert.equal(eventName, 'operation_status_changed');
```

Comment:
Are you sure that this assert will be called? Please check it, but i think that you need to send this event on EventEmitter in order to test if it was called